src/components/lookupPublicKey/useLookupPublicKey.ts

```diff
@@ -0,0 +1,117 @@
+import { ethers, UnsignedTransaction } from 'ethers';
+import { TransactionResponse } from '@ethersproject/providers';
+import { useState } from 'react';
+import axios from 'axios';
+import { useProvider, useNetwork } from 'wagmi';
+
+export enum LookupPublicKeyStatus {
+  SUCCESS,
+  ERROR,
+  NO_TRANSACTIONS,
+  LOADING,
+  INVALID_ADDRESS,
+  WRONG_NETWORK,
+  NOT_CONNECTED,
+}
+
+const etherscanEndpoint = 'https://api.etherscan.io/api';
+const etherscanApikey = process.env.REACT_APP_ETHERSCAN_APIKEY;
+const getParameters =
+  'module=account&action=txlist&startblock=0&endblock=99999999&page=1&offset=1000&sort=asc';
+
+export function useLookupPublicKey() {
```

Comment:
Consider adding some error handling to this function (and functions inside of it) -- there's a few places where things could go awry in here

Reply:
yup, thanks for point that out, added with the refactor